### ISSUE TITLE
Alerting - False alerts and Missing alerts

### ISSUE THREAD

**Issue by parikhravish** [prometheus/alertmanager#904]
What did you do?
Alert generated for an event(Label sets). Followed by this the event repeated a few times, However since prometheus generates the alert id based on the label sets it made a determination that this was a continuation alert vs new alert and due to the repeat interval no new alert was sent.
What did you expect to see?
Alert sent whenever the event repeats it self, not when it is a continuation.
What did you see instead? Under which circumstances?
New event did not generate a new alert because the label sets were the same.
Environment
Docker

System information:
Darwin ML07DTSG15KG8WL 15.6.0
insert output of uname -srm here

Prometheus version:
1.7.1
insert output of prometheus -version here

Alertmanager version:
0.7.1
insert output of alertmanager -version here (if relevant to the issue)

Prometheus configuration file:

insert configuration here
Alertmanager configuration file:
insert configuration here (if relevant to the issue)
Logs:
insert Prometheus and Alertmanager logs relevant to the issue here

Yes so for example in my case following is the situation:

Service A Working fine - No Alert
Service A Goes down - Alert issued (Taking into account the various intervals)
Service A Remains Down - No Alert Issued(Due to repeat Interval, Assuming this happened within the repeat interval)
Service A comes Back up - No Alert
Service A goes back down - No Alert Issued(Due to repeat Interval, Assuming this happened within the repeat interval) - This is where I was expecting an alert. Seems like the alert id is generated based on the label sets and alert manager thinks of this specific event as a continuation of the first instance of this event.

Basically I am trying to differentiate between an event continuing vs new event for the same service(Same label sets.)

**Comment by mxinden** [comment]
@parikhravish Can you post your Alertmanager configuration, so we can better debug this issue?

What is the time frame between A *goes down* and A *goes back down*?
Depending on your `resolve_timeout` Alertmanager never thought of *Service A* to be back up. Thereby it did not send an Alert once *Service A* went back down again.

**Comment by parikhravish** [comment]
@mxinden so for example 
Repeat Interval = 24 hours
Service A goes 8:00 AM EST
 - Alert goes out at 8:00 AM EST
Service A comes back up 9:00 AM EST
Service A goes back down at 10:00 AM EST
 - No Alert is sent out
Service A stays down for 24 hours (Next day 10:00 AM EST)
 - Alert goes out at 8:00 AM EST Next day.

Based on our observation alert id is generated based on the label sets and since its the same set of labels in the case about it is not able to differentiate and read this as a new instance of the same alert.

In the case above I would expect 3 alerts to go out: 8:00 AM EST, 10:00 AM EST and 8:00 AM EST(Next Day).

**Comment by mxinden** [comment]
Is it possible for you to post your full Alertmanager config here, so we can better debug this issue? (Of course without any secrets)

There should have been a *resolved* notification at 9:05 a.m. assumed your *resolve_timeout* is 5 minutes. The fact that the label set of the Alert does not change should not be relevant here.

@parikhravish Are you able to reproduce this issue or is this a one time occurrence? Thanks for bearing with us!

**Comment by parikhravish** [comment]
---
global:
  smtp_smarthost: "smtpe.bnymellon.net:25"
  smtp_from: "test@bnymellon.com"
  smtp_auth_username: "root"
  smtp_auth_password: "<hidden>"
  resolve_timeout: 5m
templates:
- "/etc/alertmanager/template/*.tmpl"
route:
  group_by:
  - "alertname"
  - "clustername"
  - "job"
  group_wait: "30s"
  group_interval: "5m"
  repeat_interval: "5m"
  receiver: "Ravish"
  routes:
  - match_re:
      alertname: "MonitoringNode_Down"
    receiver: "Ravish"
inhibit_rules:
- source_match:
    severity: "critical"
  target_match:
    severity: "warning"
receivers:
- name: "Ravish"
  email_configs:
  - to: "ravish.parikh@bnymellon.com"
    require_tls: false
    headers:
      subject: "{{ .GroupLabels.clustername }} - {{ .GroupLabels.job }} - {{ .GroupLabels.alertname }}"

@mxinden I am able to reproduce this issue. The alert manager config posted above was from my test environment. Where the same situation arised for repeat interval 5 minutes. So anytime an alert is reproduced multiple times within the 5 minute period the notification would not be sent.

**Comment by niravpat** [comment]
@parikhravish we have same exact issue.

**Comment by parikhravish** [comment]
any update guys? @mxinden

**Comment by mxinden** [comment]
@parikhravish I am sorry but I don't have a lot of time at the moment to look further into this. In case you have time you could write up a step by step reproduction scenario via the webhook [1] and the send-alerts script [2].

[1] https://github.com/prometheus/alertmanager/tree/master/examples/webhook
[2] https://github.com/prometheus/alertmanager/blob/master/examples/ha/send_alerts.sh

**Comment by simonpasquier** [comment]
@parikhravish do you still see the issue with the latest AlertManager release?

**Comment by TheMeier** [comment]
This issue is very outdated, therefore I am closing it. If you believe this issue is still relevant, please feel free to reopen it or create a new one.
For further support please use the [community support](https://prometheus.io/community/) channels.
